### ISSUE TITLE
Make `launch_time` a coordinate to be easier to filter the dataset

### ISSUE THREAD

**Issue by tmieslinger** [atmdrops/pydropsonde#63]
This is a request by Bjorn: he wants to have `launch_time` as a coordinate.
I do remember having this discussion. I would hope that we can collect the arguments in this issue again and come up with a text snippet for the documentation to clarify it for the future.

**Comment by hgloeckner** [comment]
I think [this halodrops issue](https://github.com/Geet-George/halodrops/issues/127) is related and explains, why it should not be a dimension (which is not what was asked, I know)

Making it a coordinate would be possible - but it's also easy for people to do that themselves ([assign_coords](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.assign_coords.html)). Filtering in a dataset is also possible with [xr.where](https://docs.xarray.dev/en/stable/generated/xarray.where.html), even if it's not made a coordinate.

**Comment by bjorn-stevens** [comment]
I had posted this in the wrong repo...

I think it would be helpful if there were physical dimensions or at least coordinates. 

I think a typical use case would be to search the dataset for sondes in a particular place at a particular time.  This is what I finally managed to make work with the present dimensions.  It seems too cumbersome to me:

```python
x1 = (ds_l3.assign_coords(time=ds_l3.launch_time)
      .swap_dims({"sonde_id":"time"})
      .sortby("time")
      .sel(time=slice("2024-08-11T00:00","2024-08-11T23:59:59"))
      )
x2 =  (x1.assign_coords(aircraft_latitude=x1.aircraft_latitude)
       .swap_dims({"time":"aircraft_latitude"})
       .sortby("aircraft_latitude")
       .sel(aircraft_latitude=slice(5.,8.))
      )
```

Having coordinates corresponding to l_time, l_lat, l_lon, l_hgt, where "l_" denotes values at first valid time, or launch, would make finding the sondes I wanted a whole lot easier.  For instance l_hgt would differentiate up from dn sondes.    Not all could be ordered, but at least time could be.
